tests/cpp_compat_tests.rs

```diff
@@ -0,0 +1,48 @@
+#![cfg(feature = "rust")]
```

Comment:
you need both `cpp` and `rust` features:

```suggestion
#![cfg(all(feature = "rust", feature = "cpp"))]
```

Reply:
Done.

Reply:
technically it will break if anyone compiles this with `default-features = false` (which is a fairly common way to compile)